### ISSUE TITLE
No need to have a ban on block-level rules creation in RucioInjector

### ISSUE THREAD

**Issue by amaltaro** [dmwm/WMCore#9654]
**Impact of the new feature**
WMAgent

**Is your feature request related to a problem? Please describe.**
After the discussion we had today about the Rucio migration. It looks like there is no need to filter out a list of datatiers from a possible block-level rule creation. Which is contrary to what we have recently implemented through this parameter (both for block and container level):
https://github.com/dmwm/WMCore/blob/master/src/python/WMComponent/RucioInjector/RucioInjectorPoller.py#L84

**Describe the solution you'd like**
Only container-level rules have to be filtered out according to the list of datatiers configured to be skipped (initially NANOAOD/NANOAODSIM).

In other words, RucioInjector should **always** create a block-level rule, provided the block has been inserted into Rucio (not in PhEDEx).

**Describe alternatives you've considered**
none

**Additional context**
none

**Comment by nsmith-** [comment]
To clarify, I think we _need_ block-level rules to be created for every output that is injected into Rucio. For container-level rules, we can relax the timeline for those if it speeds up deployment, but we also could accept them at this point. To decide, what container-level rule(s) would the current system create on an example output dataset?

**Comment by amaltaro** [comment]
Thanks, Nick! In short:
* block-level rules *must* be created for whatever data injected into Rucio by WMAgent
* container-level rules can be created, or not, doesn't matter right now.

About the container-level rules, here is the current logic/parameters (basically against a specific RSE):
https://github.com/dmwm/WMCore/blob/master/src/python/WMComponent/RucioInjector/RucioInjectorPoller.py#L419-L422

About the block-level rules, these are the current parameters implemented (basically against a specific RSE):
https://github.com/dmwm/WMCore/blob/master/src/python/WMComponent/RucioInjector/RucioInjectorPoller.py#L273-L276

**Comment by nsmith-** [comment]
Ok, one thing that I would really appreciate is that we set the `activity="Production Output"` argument in the rule creation (for both container and block rules)

I know I've asked before, but can you remind how the site for the container-level rule is chosen? Is it one of, or all of, the `NonCustodialSites` in the request?
Based on this, I would put a preference towards not making container-level rules now and waiting to do that until we finalize #9639

**Comment by amaltaro** [comment]
Okay for the "activity" bit. I see it's one of the supported parameters for the client `add_replication_rule` api.

About the RSE selection for the container-level, it would be for every single site in the `NonCustodialSites` list (even though, I don't think we ever provide multiple sites for it).

**Comment by nsmith-** [comment]
Ok, then lastly I want to check if @todor-ivanov or @sharad1126 see any issue from Unified with WMAgent & Rucio ignoring the `NonCustodialSites` (& custodial as well) for Rucio-managed NanoAOD. If not, I would suggest we remove the container rule for now.

**Comment by amaltaro** [comment]
Going through this issue once again. There seems to have two changes to be made:
* activity for rule creation (already inserted into https://github.com/dmwm/WMCore/pull/9666)
* removal of the block-level rule filter of datatiers; those should **always** get created

I might separate those black/white list of datatiers, creating specific ones for the PhEDExInjector.

Am I missing anything else?

**Comment by nsmith-** [comment]
Looks good, I think that's it for now.
